services/src/main/java/org/keycloak/email/freemarker/FreeMarkerEmailTemplateProvider.java

```diff
@@ -217,13 +219,19 @@ protected EmailTemplate processTemplate(String subjectKey, List<Object> subjectA
                 textBody = freeMarker.processTemplate(attributes, textTemplate, theme);
             } catch (final FreeMarkerException e) {
                 textBody = null;
+                logger.warn("Failed to template plain text email.", e);
             }
             String htmlTemplate = String.format("html/%s", template);
             String htmlBody;
             try {
                 htmlBody = freeMarker.processTemplate(attributes, htmlTemplate, theme);
             } catch (final FreeMarkerException e) {
                 htmlBody = null;
+                logger.warn("Failed to template html email.", e);
```

Comment:
Same as above:

`throw new EmailException("Failed to template html email.", e);`

Reply:
The same above, but vice-versa, I prefer to receive an email in plain text if html is broken.

The exception is raised only if both fail.

Reply:
Understand